qctrlopencontrols/driven_controls/driven_control.py

```diff
@@ -425,21 +390,28 @@ def _export_to_qctrl_expanded_format(
             'cylindrical', 'cartesian'; defaults to 'cylindrical'
         """
 
-        control_info = self._qctrl_expanded_export_content(
-            file_type=file_type, coordinates=coordinates
-        )
+        control_info = self._qctrl_expanded_export_content(coordinates=coordinates)
         if file_type == FileType.CSV.value:
-            with open(filename, "wt") as handle:
-
-                control_info = "\n".join(control_info)
-                handle.write(control_info)
+            _ = control_info.pop("name")
+            control_info["maximum_rabi_rate"] = [
+                self.maximum_rabi_rate
+            ] * self.number_of_segments
+            field_names = sorted(control_info.keys())
+            with open(filename, "w", newline="") as file:
+                writer = csv.DictWriter(file, fieldnames=field_names)
+                writer.writeheader()
+
+                for index in range(self.number_of_segments):
+                    writer.writerow(
+                        {name: control_info[name][index] for name in field_names}
+                    )
         else:
             with open(filename, "wt") as handle:
                 json.dump(control_info, handle, sort_keys=True, indent=4)
 
     def export_to_file(
         self,
-        filename=None,
+        filename,
```

Comment:
Update docstring

Reply:
Sorry, did you mean the doc for `filename`?
Seems it was not optional in the doc.

Reply:
D'oh, sorry. I just assumed it was.